streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -483,34 +577,101 @@ public StreamThread(final InternalTopologyBuilder builder,
             log.info("{} Custom offset resets specified updating configs original auto offset reset {}", logPrefix, originalReset);
             consumerConfigs.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "none");
         }
+        this.consumer = clientSupplier.getConsumer(consumerConfigs);
+        taskManager.setConsumer(consumer);
+    }
+
+    @SuppressWarnings("ConstantConditions")
```

Comment:
it is to supress the warning caused by this line:
`final int rebalanceTimeoutMs = (Integer) ConfigDef.parseType(ConsumerConfig.MAX_POLL_INTERVAL_MS_CONFIG, maxPollInterval, Type.INT);`